app/src/main/java/io/pumpkinz/pumpkinreader/NewsDetailActivity.java

```diff
@@ -33,12 +54,43 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
-    private void setUpToolbar() {
-        Toolbar appBar = (Toolbar) findViewById(R.id.app_bar);
-        setSupportActionBar(appBar);
+    private void setUpViewPager(ViewPager viewPager) {
+        NewsDetailViewPagerAdapter adapter = new NewsDetailViewPagerAdapter(getSupportFragmentManager());
+
+        adapter.addFragment(getResources().getString(R.string.link), new WebViewFragment());
+        adapter.addFragment(getResources().getString(R.string.comments), new NewsDetailFragment());
+
+        viewPager.setAdapter(adapter);
+    }
+
+    private void setUpTab(TabLayout tabLayout, ViewPager viewPager) {
+        tabLayout.setupWithViewPager(viewPager);
+
+        if (news.getUrl() == null || news.getUrl().isEmpty()) {
+            tabLayout.getTabAt(TAB_COMMENTS_IDX).select();
+        }
+    }
+
+    private void setUpFAB(FloatingActionButton fab) {
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent shareIntent = new Intent(Intent.ACTION_SEND);
+
+                String text;
```

Comment:
May be we should add the title?

OH, and what about "Shared from Pumpkin Reader for Hacker News. Download Pumpkin Reader for free here"?